vcd/resource_vcd_vapp_vm.go

```diff
@@ -680,13 +686,16 @@ func resourceVcdVAppVmDelete(d *schema.ResourceData, meta interface{}) error {
 		}
 
 		log.Printf("[TRACE] Powering on vApp: %s", vapp.VApp.Name)
-		task, err = vapp.PowerOn()
-		if err != nil {
-			return fmt.Errorf("error Powering on vApp: %#v", err)
-		}
-		err = task.WaitTaskCompletion()
+		err = retryCall(vcdClient.MaxRetryTimeout, func() *resource.RetryError {
```

Comment:
Please add a comment on why this retry is needed.

BTW, for the future we should seriously think about encapsulating such behaviour inside the vapp.PowerOn() function, so that all callers benefit from it.

Reply:
This isn't PowerOn() issue it's global vApp issue. We will need some kind vApp scoped lock.